### PR TITLE
Rearrange tutorial index

### PR DESCRIPTION
**Pull request by johnwhitington**
This PR reduces the length of the tutorials index page by about one quarter, and rearranges its contents.

- removes archived tutorials
- reorders list of tutorials roughly in order of complexity / expertise
- external links section reduced to fewer subheadings by moving things around and renaming sections
- cleans up language for tone throughout

Should the PR be accepted in principle, I will update all the other language index files, at least to the extent possible to an English speaker.

### CODE REVIEW

**Comment by johnwhitington** [comment]
(This is a prelude to lots of work I want to do on the tutorials. There may be further clean-up in the future, but I wanted to clean it up somewhat before beginning.)

File: site/learn/tutorials/index.md [hunk 1]
```diff
@@ -3,11 +3,11 @@
 
 # OCaml tutorials
 
-####  Your Help is Needed
-Many of the tutorials below need updating and tutorials on many new
-topics are needed. Please contribute by visiting this project's repo on
-[GitHub](https://github.com/ocaml/ocaml.org); you may use the issue
-tracker there to request or offer new tutorials. Thanks!
+#### Your Help is Needed
+Some of the tutorials above need updating and tutorials on new topics are
```
Comment (avsm) [inline site/learn/tutorials/index.md]: How is this `above`? The list of tutorials is still below this paragraph as far as I can tell :-)

**Comment by avsm** [comment]
This can be viewed on https://staging.ocaml.org/learn/tutorials/ now

**Comment by johnwhitington** [comment]
Thanks! I'll have a careful look tomorrow morning to see if there are any other howlers.

File: site/learn/tutorials/index.md (hunk 1 above)
Comment (johnwhitington) [reply site/learn/tutorials/index.md]: Fixed. I remember that using "above" and "below" is considered bad writing (and is unnecessary here anyway) so I use "these tutorials".

**Comment by johnwhitington** [comment]
I will propose changes to the other language files today, then I will be finished.

**Comment by johnwhitington** [comment]
Ok, I'm finished.

Most of the other languages were very different, and I was not confident to fix them. So, I contented myself with removing from them the out-of-date links we have removed from the English one. So they are at least tidier.

This can be merged now if you are happy with it.

**Comment by avsm** [comment]
Many thanks @johnwhitington.

**Pull request merged**